### PR TITLE
fix updateMetadata with new Metadata DAO;

### PR DESCRIPTION
**Pull request by jesuspg**
@hmunfru @jframos

### CODE REVIEW

File: core/src/main/java/com/telefonica/euro_iaas/sdc/manager/impl/ProductManagerImpl.java [hunk 1]
```diff
@@ -146,6 +146,34 @@ public Product insert(Product product, String tenantId) throws AlreadyExistsEnti
         return result;
     }
 
+    @Override
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/sdc/manager/impl/ProductManagerImpl.java]: please, add javadoc to the new method

File: core/src/main/java/com/telefonica/euro_iaas/sdc/manager/impl/ProductManagerImpl.java [hunk 2]
```diff
@@ -146,6 +146,34 @@ public Product insert(Product product, String tenantId) throws AlreadyExistsEnti
...
+            metadata = product.getMetadata(metadataName);
+            metadata = metadataDao.loadById(metadata.getId());
+        } catch (Exception e) {
+            log.warning("Exception: metadata not found or wrong" + metadataName);
+            throw new EntityNotFoundException(Metadata.class, "Metadata not found : " + metadataName, null);
+        }
+        return metadata;
+    }
+
+    @Override
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/sdc/manager/impl/ProductManagerImpl.java]: please, add javadoc to the new method

File: core/src/main/java/com/telefonica/euro_iaas/sdc/manager/impl/ProductManagerImpl.java [hunk 3]
```diff
@@ -146,6 +146,34 @@ public Product insert(Product product, String tenantId) throws AlreadyExistsEnti
...
+
+        Product product;
+        Metadata metadata;
+        try {
+            product = this.load(productName);
+        } catch (Exception e) {
+            log.warning("The product: " + productName + " does not exist" + e.getMessage());
+            throw new EntityNotFoundException(Product.class, productName, null);
+        }
+        try {
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/sdc/manager/impl/ProductManagerImpl.java]: I suggest to put a blank line before try and delete the blank file after try {

File: core/src/test/java/com/telefonica/euro_iaas/sdc/dao/impl/MetadataDaoJpaImplTest.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import com.telefonica.euro_iaas.commons.dao.AlreadyExistsEntityException;
+import com.telefonica.euro_iaas.commons.dao.EntityNotFoundException;
+import com.telefonica.euro_iaas.sdc.dao.MetadataDao;
+import com.telefonica.euro_iaas.sdc.model.Metadata;
+
+/**
+ *
```
Comment (flopezag) [inline core/src/test/java/com/telefonica/euro_iaas/sdc/dao/impl/MetadataDaoJpaImplTest.java]: More verbose... :-1:

File: core/src/test/java/com/telefonica/euro_iaas/sdc/dao/impl/MetadataDaoJpaImplTest.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+ * 
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(locations = { "classpath:/spring-test-db-config.xml", "classpath:/spring-dao-config.xml" })
+public class MetadataDaoJpaImplTest {
+
+    @Autowired
+    private MetadataDao metadataDao;
+
+    @Test
```
Comment (flopezag) [inline core/src/test/java/com/telefonica/euro_iaas/sdc/dao/impl/MetadataDaoJpaImplTest.java]: I would like to have a more description information on each test method in order to know what are you doing.

File: core/src/test/java/com/telefonica/euro_iaas/sdc/dao/impl/MetadataDaoJpaImplTest.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        Metadata responseMetadata = metadataDao.loadById(metadata.getId());
+
+        // then
+        assertNotNull(responseMetadata);
+        assertEquals("key1", responseMetadata.getKey());
+        assertEquals("value1", responseMetadata.getValue());
+        assertEquals("description", responseMetadata.getDescription());
+    }
+
+    @Test
```
Comment (flopezag) [inline core/src/test/java/com/telefonica/euro_iaas/sdc/dao/impl/MetadataDaoJpaImplTest.java]: I would like to have a more description information on each test method in order to know what are you doing.

File: core/src/test/java/com/telefonica/euro_iaas/sdc/dao/impl/MetadataDaoJpaImplTest.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        Metadata responseMetadata = metadataDao.merge(updatedMetadata);
+
+        // then
+        assertNotNull(responseMetadata);
+        assertEquals("key2", responseMetadata.getKey());
+        assertEquals("newValue", responseMetadata.getValue());
+        assertEquals("description", responseMetadata.getDescription());
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
```
Comment (flopezag) [inline core/src/test/java/com/telefonica/euro_iaas/sdc/dao/impl/MetadataDaoJpaImplTest.java]: I would like to have a more description information on each test method in order to know what are you doing.

**Comment by flopezag** [comment]
LGTM

**Comment by hmunfru** [comment]
LGTM

**Comment by jframos** [comment]
LGTM
(To be tested in the Experimentation Environment)

**Pull request merged**